Umap polish (#423)
AlexsLemonade/refinebio-examples: 02-microarray/dimension-reduction_microarray_02_umap.Rmd

# Read in data TSV file
df <- readr::read_tsv(data_file) %>%
  # Tuck away the gene ID  column as rownames, leaving only numeric values
  tibble::column_to_rownames("Gene")
```

Let's ensure that the metadata and data are in the same sample order. 

```{r}

> # Make the sure the columns (samples) are in the same order as the metadata

Reviewer: ```suggestion
# Make sure the columns (samples) are in the same order as the metadata
```

df <- df %>%
  dplyr::select(metadata$geo_accession)

# Check if this is in the same order
all.equal(colnames(df), metadata$geo_accession)
```

Now we are going to use a combination of functions from the `umap` and the `ggplot2` packages to perform and visualize the results of the Uniform Manifold Approximation (UMAP) dimension reduction technique on our medulloblastoma samples.

## Perform Uniform Manifold Approximation (UMAP)

In this code chunk, we are going to perform Uniform Manifold Approximation (UMAP) on our data and create a data frame using the UMAP scores and the variables from our metadata that we are going to use to annotate our plot later.
The `umap()` function calculates scores for each row of a matrix, but our data is arranged with each sample in a column, so we will need to transpose the data frame first.

```{r}
# Perform UMAP using the `umap::umap()` function
umap_data <- umap::umap(
  t(df) # transpose our data frame to obtain PC scores for samples, not genes
)

# Make into data frame for plotting with `ggplot2`

> # The umap values we need for plotting are stored in the `layout` element

Reviewer: ```suggestion
# The UMAP values we need for plotting are stored in the `layout` element
```

Just style reasons.

umap_df <- data.frame(umap_data$layout) %>%

>   # Turn sample IDs stored as row names into column

Reviewer: ```suggestion
  # Turn sample IDs stored as row names into a column
```

Our histological subtype groups don't appear to be clustering in a discernible pattern. 
We could test out other variables as annotation labels to get a further understanding of the cluster behavior of each subgroup.

> ### Interpretation of UMAP plot and results

Reviewer: Did you not want to also add this section to the RNA-seq version? (or was it already there and I can't see it)

Author: Yup, it is already there: copied it directly!

> 1. Note that the coordinates of UMAP output for any given cell can change dramatically depending on parameters, and even run to run with the same parameters  (Also why setting the seed is important).

Reviewer: ```suggestion
1. Note that the coordinates of UMAP output for any given cell can change dramatically depending on parameters, even if ran with the same parameters (Also why setting the seed is important).
<!-- TODO: set the seed glossary link --> 
```
